EmailValidator/Validation/MultipleValidationWithAnd.php

```diff
@@ -71,8 +70,11 @@ public function isValid($email, EmailLexer $emailLexer)
                 break;
             }
         }
-        $this->error = new MultipleErrors($errors);
-        
+
+        if (!empty(array_filter($errors))) {
```

Comment:
We can instead filter them before adding them to `$errors`, replacing `$errors[] = $validation->getError();` with something like this:
```php
//remove $errors = [];
//...
foreach ($this->validations as $validation) {
    //...
    $errors = $this->addNewError($validation->getError(), $errors);
    // ...
}
//...
private function addNewError($possibleError, array $errors) {
    if(null !== $possibleError) {
        $errors[] = $possibleError;
    }
    return $errors;
}
```
I propose this way because I'm not a fan of passing by reference, so I prefer a function that returns the new array.
Also the error list is being augmented only when necessary.

What do you think?